### PR TITLE
[java] Optimize typeresolution, by skipping package and import declarations in visit(ASTName)

### PR DESCRIPTION
**Pull request by WinterGrascph**

### CODE REVIEW

**Comment by jsotuyod** [comment]
It seems I was not clear enough, I'm really sorry.

The idea is to have the visitor (which uses DFS) skip visiting the children of `ASTImportDeclaration` and `ASTPackageDeclaration` altogether, so you never get to `visit(ASTName node, Object data)` having those as parent of `node`.

To do this, simply make sure that both `visit(ASTImportDeclaration node, Object data)` and `visit(ASTPackageDeclaration node, Object data)` never call `super.visit(node, data)`, since that's where child visitation is performed.

By doing this, you should never worry about handling `ASTName` nodes for imports and packages.

**Comment by WinterGrascph** [comment]
Yep, your version is a much simpler solution.

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java [hunk 1]
```diff
@@ -181,8 +181,14 @@ public Object visit(ASTCompilationUnit node, Object data) {
     }
 
     @Override
+    public Object visit(ASTPackageDeclaration node, Object data) {
+        return data;
```
Comment (jsotuyod) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java]: I wouldn't mind having a comment here explaining why we don't call super

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java [hunk 2]
```diff
@@ -192,6 +198,7 @@ public Object visit(ASTImportDeclaration node, Object data) {
         if (node.getType() != null) {
             node.setPackage(node.getType().getPackage());
         }
+
         return data;
```
Comment (jsotuyod) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java]: ditto

**Comment by jsotuyod** [comment]
Merged already! Keep it up!

**Pull request merged**